### PR TITLE
planner: adapter the generic hint struct TableOptimizerHint

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

The https://github.com/pingcap/parser/pull/747 introduces a new field `HintData` into the struct `TableOptimizerHint`, which is used to save the extra hint information. The former implementation will cause field count growth as we adding more hint types

### What is changed and how it works?

1. Update the deps.
2. Fix the compiler errors.

### Check List 

Tests 

 - Unit test
 - Integration test

Release note

 - No need.

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -2240,10 +2240,10 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint, nodeType n
...
 			}
 		case HintReadFromStorage:
-			if hint.StoreType.L == HintTiFlash {
+			if hint.HintData.(model.CIStr).L == HintTiFlash {
 				tiflashTables = tableNames2HintTableInfo(b.ctx, hint.Tables, b.hintProcessor, nodeType, currentLevel)
 			}
-			if hint.StoreType.L == HintTiKV {
+			if hint.HintData.(model.CIStr).L == HintTiKV {
 				tikvTables = tableNames2HintTableInfo(b.ctx, hint.Tables, b.hintProcessor, nodeType, currentLevel)
 			}
```
Comment (kennytm) [inline planner/core/logical_plan_builder.go]: Use a `switch` here

Comment (lonng) [reply planner/core/logical_plan_builder.go]: addressed

**Pull request merged**